draft-ietf-rats-corim.md

```diff
@@ -2149,11 +2149,21 @@ Reference Value does not match.
 
 ##### Comparison for raw-value entries
 
-> [Andy] *I think this comparison method only works if the entry is at key 4 (because
-there needs to be a mask at key 5). Should we have a Reference Value of this
-which stores `[expect-raw-value raw-value-mask]` in an array?*
+The value stored under `measurement-values-map` key 4 is a `raw-value` entry, which must have type BSTR.
```

Comment:
There is asymmetry in the definition of raw-value and raw-value-mask in that the mask is always `bstr` while raw-value-type-choice is extensible. The implication seems to be that the CBOR tag tells the Verifier how to map the raw value into a `bstr` so that the mask can be applied consistently.

If there is a way to write the text so that always applies regardless of $raw-value-type-choice extensions, then that is preferable. 

Nevertheless, before the raw value can be applied to the mask it must undergo a conversion to `bstr`. 

I think I would prefer a definition of raw-value that puts both the value and the mask behind the same CBOR tag. But what is there is OK for now IMHO.

Reply:
@andrew-draper how disruptive is it to drop raw-value-mask in favor of a tagged-masked-bytes choice in raw-value-type-choice?

```
563([data: bstr, mask: bstr])
```

Reply:
> tagged-masked-bytes = 563([data: bstr, mask: bstr])

From the perspective of moving the Intel profile to a standard representation. This approach would be easier as the Intel profile already defines something similar:
```
mask-eq = 1
mask-operator = mask-eq
mask-type = bstr
mask-expression = [ mask-operator, value: mask-type, mask: mask-type ]
tagged-mask-expression = #6.60010( mask-expression )
```
I could see the following being easier to migrate to than the current raw-value:
```
mask-expression = [ mask-operator, 563([ data: bstr, mask: bstr]) ]
tagged-mask-expression = #6.60010( mask-expression )
```
In terms of disruptiveness, we should deprecate before deleting. Maybe the corim spec could add a second masked value type while discouraging use of raw-value?

Reply:
@andrew-draper : Add more explanation to explain how to use the `raw-value-type`

Reply:
I think tagged-mask-expresssion is a slightly nicer way to do the comparison. Since the mask only appears in CoRIM files (not in evidence) I don't think this has any outside effects.
I agree with deprecating before deleting.

Reply:
The most recent proposal uses a mask expression (named `raw-value-compare`) similar to what is proposed by Ned.